.github/actions/build-release-candidate/action.yml

```diff
@@ -0,0 +1,52 @@
+name: 'Build release candidate'
+description: 'Builds and caches the release candidate that will be used for deployment and validations leading up to deployment'
+inputs:
+  app_name:
+    description: 'Name of application'
+outputs:
+  image_cache_key:
+    description: "Cache key for the .tar of the docker image"
+    value: ${{ steps.create-image-identifier.outputs.image }}
+ 
+runs:
+  using: "composite"
+  steps:        
+
+    - name: Create image identifier
+      id: create-image-identifier
+      run: |
+        IMAGE_NAME=$(make APP_NAME=${{ inputs.app_name }} release-image-name)
+        IMAGE_TAG=$(make release-image-tag)
+        echo "image=$IMAGE_NAME:$IMAGE_TAG" >> "$GITHUB_OUTPUT"
+      shell: bash
+
+    - name: Check whether the image has already been built
+      id: check-image-already-exists
+      uses: actions/cache@v4
+      with:
+        path: /tmp/docker-image.tar
+        key: ${{ steps.create-image-identifier.outputs.image }}
+        lookup-only: true
```

Comment:
what's the reason for only doing a lookup? i think we'd want to download the image if it's already built